package/MDAnalysis/analysis/msd.py

```
@@ -0,0 +1,419 @@
+        if self.fft == True:
+            self._conclude_fft()
+        else:
+            self._conclude_simple()
+
+    def _conclude_simple(self):
+        r""" Calculates the MSD via the simple "windowed" algorithm.
+
+        Parameters
+        ----------
+        self._position_array : :class:`numpy.ndarray`
+            Array of particle positions with respect to time 
+            shape (n_frames, n_particles, dim_fac).
+
+        Returns
+        -------
+        self.timeseries : :class:`numpy.ndarray`
+            The MSD as a function of lag-time.
+
+        """
+        lagtimes = np.arange(1, self.n_frames)
+        for lag in lagtimes:
+            disp = self._position_array[:-lag, :, :] - \
+                self._position_array[lag:, :, :]
+            sqdist = np.square(disp, dtype=np.float64).sum(
+                axis=-1, dtype=np.float64)  # np.float64 required
+            self.msds_by_particle[lag, :] = np.mean(
+                sqdist, axis=0, dtype=np.float64)
+        self.timeseries = self.msds_by_particle.mean(axis=1, dtype=np.float64)
+
+    def _conclude_fft(self):  # with FFT, np.float64 bit prescision required.
+        r""" Calculates the MSD via the FCA fast correlation algorithm.
+
+        Parameters
+        ----------
+        self._position_array : :class:`numpy.ndarray`
+            Array of particle positions with respect to time 
+            shape (n_frames, n_particles, dim_fac).
+
+        Returns
+        -------
+        self.timeseries : :class:`numpy.ndarray`
+            The MSD as a function of lag-time.
+
+        """
+        try:
+            import tidynamics
+        except ImportError:
+            ("""ERROR --- tidynamics was not found!
+
+                tidynamics is required to compute an FFT based MSD (default)
+
+                try installing it using pip eg:
+                    
+                    pip install tidynamics
+                    
+                or set fft=False""")
```

What is supposed to happen here? I would have expected that you raise an exception (`six.raise_from(ImportError, ...)`) with such a message. At the moment this just keeps going and then fails with a NameError (I think).

---

To clarify on what @orbeckst said here; when you catch the ImportError with `except ImportError`, you interrupt the error process. So, because you don't then `raise` another error, the code just continues. You need to raise a new error for it to die here. Additionally, this string doesn't do anything, as it is not attached to a command like `raise` or `print` etc. You probably wanted something like `raise ImportError("""ERROR...""")` instead.